File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    unordered_map<string,int> m;
    for(auto player : participant){
        if(m.end() == m.find(player)){
            m.insert(make_pair(player,1));
        }
        else{
            m[player]++;
        }
    }
    for(auto player : completion){
        m[player]--;
    }
    for(auto player : participant){
        if(m[player]> 0){
            answer = player;
            break;
        }
    }
    return answer;
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.HashMap;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        HashMap<String, Integer> person = new HashMap<>();
        
        for (String per : participant) {
            person.put(per, person.getOrDefault(per, 0) + 1);
        }
        
        for (String com : completion) {
            person.put(com, person.get(com) -1);
        }
        
        for (String key : person.keySet()) {
            if (person.get(key) > 0) {
                return key;
            }
        }
        
        
        return answer;
    }
}